Mail/Views/New Message/AutocompletionView.swift

```diff
@@ -18,48 +18,78 @@
 
 import MailCore
 import MailResources
+import RealmSwift
 import SwiftUI
 
 struct AutocompletionView: View {
+    @State private var shouldAddUserProposal = false
+
     @Binding var autocompletion: [Recipient]
-    @Binding var unknownRecipientAutocompletion: String
+    @Binding var currentSearch: String
+    @Binding var addedRecipients: RealmSwift.List<Recipient>
 
-    let onSelect: (Recipient) -> Void
+    let addRecipient: @MainActor (Recipient) -> Void
 
     var body: some View {
-        LazyVStack {
+        LazyVStack(spacing: UIConstants.autocompletionVerticalPadding) {
             ForEach(autocompletion) { recipient in
-                VStack(alignment: .leading, spacing: 8) {
-                    Button {
-                        onSelect(recipient)
-                    } label: {
-                        RecipientCell(recipient: recipient)
-                    }
-                    .padding(.horizontal, 8)
+                let isLastRecipient = autocompletion.last?.isSameRecipient(as: recipient) == true
+                let isUserProposal = shouldAddUserProposal && isLastRecipient
 
-                    IKDivider()
+                VStack(alignment: .leading, spacing: UIConstants.autocompletionVerticalPadding) {
+                    AutocompletionCell(
+                        addRecipient: addRecipient,
+                        recipient: recipient,
+                        highlight: currentSearch,
+                        alreadyAppend: addedRecipients.contains { $0.isSameRecipient(as: recipient) },
+                        unknownRecipient: isUserProposal
+                    )
+
+                    if !isLastRecipient {
+                        IKDivider()
+                    }
                 }
             }
+        }
+        .onAppear {
+            updateAutocompletion(currentSearch)
+        }
+        .onChange(of: currentSearch, perform: updateAutocompletion)
```

Comment:
To have a cohesive experience with the search, should we debounce this as well ? What do you think ?

This is what we have in the search:
` .debounce(for: .seconds(0.3), scheduler: DispatchQueue.main)`

Reply:
Seems a good idea, I will try it!

Reply:
Perfect, thanks for trying it out.